src/commands/mod.rs

```diff
@@ -298,29 +302,96 @@ impl DaemonControl {
             .coins(CoinType::All)
             // Can't use into_values as of Rust 1.48
             .into_iter()
-            .map(|(_, coin)| {
-                let Coin {
-                    amount,
-                    outpoint,
-                    block_info,
-                    spend_txid,
-                    spend_block,
-                    ..
-                } = coin;
-                let spend_info = spend_txid.map(|txid| LCSpendInfo {
-                    txid,
-                    height: spend_block.map(|b| b.height),
-                });
-                let block_height = block_info.map(|b| b.height);
-                ListCoinsEntry {
-                    amount,
-                    outpoint,
-                    block_height,
-                    spend_info,
+            .map(|(_, coin)| ListCoinsEntry::from(coin))
+            .collect();
+        ListCoinsResult { coins }
+    }
+
+    /// Select coins for spend.
+    pub fn select_coins_for_spend(
+        &self,
+        spend_amount: bitcoin::Amount,
+        spend_feerate_vb: u64,
+        long_term_feerate_vb: u64,
```

Comment:
That sounds good. If I understand well, bitcoin core uses a value based on the config option `-consolidatefeerate`, which has a default value of 10 sat/vbyte:
- https://github.com/bitcoin/bitcoin/blob/master/src/wallet/wallet.h#L87
- https://github.com/bitcoin/bitcoin/blob/master/src/wallet/spend.cpp#L842

We could follow a similar approach, giving the user the option to set their own value if they wish. I think using "consolidation" instead of "long-term" in the name would make it easier to reason about for the user when choosing a value.

Reply:
Good point. @jp1ac4 let's stick with a constant for now (we'll need one as a fallback anyways), we can implement what Loyd suggests in a follow-up.